src/vivarium/framework/values.py

```diff
@@ -24,14 +25,41 @@
 from vivarium.manager import Interface, Manager
 from vivarium.types import NumberLike
 
+if TYPE_CHECKING:
+    from vivarium.framework.engine import Builder
+
+
+class ValueSource(Protocol):
+    def __call__(self, *args: Any, **kwargs: Any) -> NumberLike:
+        ...
+
+
+class ValueMutator(Protocol):
+    def __call__(self, *args: Any, value: NumberLike, **kwargs: Any) -> NumberLike:
```

Comment:
Can we be more specific w/ typing *args and **kwargs throughout these? (I don't think we can but I haven't thought hard about it yet)

Reply:
I think so-- we can use paramSpec, but that would necessitate some other changes.
https://peps.python.org/pep-0612/#abstract